netlogo-gui/src/main/ide/ShowUsageBox.scala

```diff
@@ -0,0 +1,168 @@
+        }
+        if (dataModel.getRowCount != 0) {
+          usageTable.setDefaultRenderer(classOf[String], new LineRenderer(Some(token.text)))
+          usageTable.setPreferredScrollableViewportSize(usageTable.getPreferredSize())
+          usageTable.setFillsViewportHeight(true)
+          usageBox.setSize(usageTable.getPreferredSize)
+          usageTable.validate()
+          usageBox.setLocation(me.getLocationOnScreen)
+          usageBox.setVisible(true)
+        }
+      }
+    }
+  }
+
+  def getUsage(source: String, token: Token): Iterator[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.filter(t => t.text.equalsIgnoreCase(token.text) && t.start != token.start)
+  }
+
+  def findTokenContainingPosition(source: String, position: Int): Option[Token] = {
+    val iterator = Femto.scalaSingleton[TokenizerInterface]("org.nlogo.lex.Tokenizer").tokenizeString(source)
+    iterator.find(p => p.start < position && p.end >= position)
+  }
+
+  class LineRenderer(boldedString: Option[String]) extends TableCellRenderer {
+    override def getTableCellRendererComponent(table: JTable, value: AnyRef, isSelected: Boolean, hasFocus: Boolean, row: Int, column: Int): Component = {
+      val pane  = new JEditorPane()
+      pane.setOpaque(true)
+      pane.setBorder(BorderFactory.createEmptyBorder(1,0,0,0))
+      val editorKit = boldedString match {
+        case None => new HighlightEditorKit(null, editorArea.colorizer)
+        case Some(selectedString) => new BoldEditorKit(selectedString)
+      }
+      pane.setEditorKit(editorKit)
+      pane.setText(value.asInstanceOf[String])
+      val alternate = UIManager.getColor("Table.alternateRowColor")
+      val defaults = new UIDefaults()
+      if (row%2 == 0){
+        defaults.put("EditorPane[Enabled].backgroundPainter", Color.WHITE)
+        pane.setBackground(Color.WHITE)
+      }
+      else {
+        defaults.put("EditorPane[Enabled].backgroundPainter", alternate)
+        pane.setBackground(alternate)
+      }
+      if(isSelected) {
+        pane.setBackground(usageTable.getSelectionBackground)
+        pane.setForeground(usageTable.getSelectionForeground)
+      }
+      pane.putClientProperty("Nimbus.Overrides", defaults)
+      pane.putClientProperty("Nimbus.Overrides.InheritDefaults", true)
+      pane
+    }
+  }
+  class LineNumberRenderer extends DefaultTableCellRenderer {
+    override def setValue(value: AnyRef) = {
+      setText(editorArea.offsetToLine(value.asInstanceOf[Token].start).toString)
+    }
+  }
+  import org.nlogo.editor.{ HighlightEditorKit, HighlightView }
+  class BoldEditorKit(selectedString: String) extends HighlightEditorKit(null, editorArea.colorizer) {
+    import scala.collection.immutable.Range
+    class BoldView(elem: javax.swing.text.Element) extends HighlightView(this.pane, elem, this.colorizer) {
+      var boldingRange = Option.empty[Range]
```

Comment:
I made this as a proof-of-concept, but this should detect when there are multiple occurrences of the token within a line and bold all of them. You could represent this with a `Seq[Range]`.